src/Chainweb/BlockHeaderDB.hs

```diff
@@ -255,7 +168,11 @@ pruneForks logger cdb limit callback = do
 
     !roots <- keys cdb Nothing Nothing
         (Just $ MinRank $ Min rootHeight)
-        (Just $ MaxRank $ Max rootHeight)
+        Nothing
+            -- include all block headers in the root set, so that all headers
+            -- are included in the marking phase. PayloadBlockHashes can be
+            -- shared between blocks at different height, so we must make sure
+            -- that they are retained if a root depends on them.
```

Comment:
How is this related to the PR? Again, isn't this on master already?

Reply:
Emmanuel's branch had a bad merge and this branch is relative to that. (Stu wanted to compare my changes to Emmanuel's that were already reviewed). If you want to diff against master I can pull up a different PR for that.